src/torcontrol.cpp

```diff
@@ -481,6 +483,53 @@ TorController::~TorController()
     }
 }
 
+void TorController::get_socks_cb(TorControlConnection& _conn, const TorControlReply& reply)
+{
+    // NOTE: We can only get here if -onion is unset
+    std::string socks_port_str;
+    if (reply.code == 250) {
+        for (const auto& line : reply.lines) {
+            if (0 == line.compare(0, 20, "net/listeners/socks=")) {
+                const std::string port_list_str = line.substr(20);
+                std::vector<std::string> port_list;
+                boost::split(port_list, port_list_str, boost::is_any_of(" "));
+                for (auto& portstr : port_list) {
+                    if (portstr.empty()) continue;
+                    if ((portstr[0] == '"' || portstr[0] == '\'') && (*portstr.rbegin() == '"' || *portstr.rbegin() == '\'')) {
+                        portstr = portstr.substr(1, portstr.size() - 2);
+                        if (portstr.empty()) continue;
+                    }
+                    socks_port_str = portstr;
+                    if (0 == portstr.compare(0, 10, "127.0.0.1:")) {
+                        // Prefer localhost - ignore other ports
+                        break;
+                    }
+                }
+            }
+        }
+        if (!socks_port_str.empty()) {
+            LogPrint(BCLog::TOR, "tor: Get SOCKS port command yielded %s\n", socks_port_str);
+        } else {
+            LogPrintf("tor: Get SOCKS port command returned nothing\n");
+        }
+    } else if (reply.code == 510) {  // 510 Unrecognized command
+        LogPrintf("tor: Get SOCKS port command failed with unrecognized command (You probably should upgrade Tor)\n");
```

Comment:
Do you know what is the minimum version of Tor supporting this query? is it newer or older than what we already require?

Reply:
> Do you know what is the minimum version of Tor supporting this query? is it newer or older than what we already require?
> (not that it matters much, as I se it'll gracefully fall back to the old behavior; but it might be useful for testing)

Tyi: This info call /net/listeners/socks is according to tor doku possible since [ 0.2.2.26-beta.]
I tested with versions 0.2.9.16 and 0.4.3.5

Changes in tor version 0.2.2.32 - 2011-08-27

>   o Minor features (controller interface):
    - ****New "GETINFO net/listeners/(type)"** controller command to return
      a list of addresses and ports that are bound for listeners for a**
      given connection type. This is useful when the user has configured
      "SocksPort auto" and the controller needs to know which port got
      chosen. Resolves another part of ticket 3076.
    - Have the controller interface give a more useful message than
      "Internal Error" in response to failed GETINFO requests.